test/utils/sentryeventemitterfallback.test.ts

```diff
@@ -0,0 +1,241 @@
+import { logger } from '@sentry/utils';
+import { DeviceEventEmitter } from 'react-native';
+
+import { NewFrameEventName } from '../../src/js/utils/sentryeventemitter';
+import { createSentryFallbackEventEmitter } from '../../src/js/utils/sentryeventemitterfallback';
+
+// Mock dependencies
+
+jest.mock('react-native', () => {
+  return {
+    DeviceEventEmitter: {
+      addListener: jest.fn(),
+      emit: jest.fn(),
+    },
+    Platform: {
+      OS: 'ios',
+    },
+  };
+});
+
+jest.mock('../../src/js/utils/environment', () => ({
+  isTurboModuleEnabled: () => false,
+}));
+
+jest.mock('../../src/js/wrapper', () => jest.requireActual('../mockWrapper'));
+
+jest.mock('@sentry/utils', () => ({
+  logger: {
+    log: jest.fn(),
+    error: jest.fn(),
+  },
+}));
+
+import { NATIVE } from '../../src/js/wrapper';
+
+describe('SentryEventEmitterFallback', () => {
+  let emitter: ReturnType<typeof createSentryFallbackEventEmitter>;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // @ts-expect-error test
+    jest.spyOn(window, 'requestAnimationFrame').mockImplementation(cb => cb());
+    emitter = createSentryFallbackEventEmitter();
+    NATIVE.getNewScreenTimeToDisplay = jest.fn(() => Promise.resolve(12345));
+  });
+
+  afterEach(() => {
+    // @ts-expect-error test
+    window.requestAnimationFrame.mockRestore();
+    NATIVE.getNewScreenTimeToDisplay = jest.fn();
+  });
+
+  it('should initialize and add a listener', () => {
+    emitter.initAsync();
+
+    expect(DeviceEventEmitter.addListener).toHaveBeenCalledWith(NewFrameEventName, expect.any(Function));
+  });
+
+  it('should start listener and use fallback when native call returned undefined/null', async () => {
+    jest.useFakeTimers();
+    const fallbackTime = Date.now() / 1000;
+
+    const animation = NATIVE.getNewScreenTimeToDisplay as jest.Mock;
+    animation.mockReturnValue(Promise.resolve());
+
+    emitter.startListenerAsync();
+    await animation; // wait for the Native execution to be completed.
```

Comment:
The current `await animation` allows the next event loop, but doesn't wait for the execution to be completed. I would rename this to be more accurate, but the test works correct in both cases.

```suggestions
    (NATIVE.getNewScreenTimeToDisplay as jest.Mock).mockReturnValue(Promise.resolve());

    emitter.startListenerAsync();

    // Wait for the next event loop to allow startListenerAsync to call NATIVE.getNewScreenTimeToDisplay
    await Promise.resolve();
```

Reply:
Thank you for the suggestion! I will apply it to the other tests too.